README.md

```diff
@@ -1,16 +1,1837 @@
+
+All of the [Invocation]s of the `Effect` MUST be concurrent with one another, unless explicitly arranged differently using [Pipelines].
+
+Effect MAY instruct the [Executor] to run set of [Task] [Invocation]s in concurrent execution threads by providing them as list under `fork` field.
+
+Effect MAY instruct the [Executor] that the [Task] [Invocation] is a continuation of the current execution thread by providing it under `join` field.
+
+Often [Invocation] in the `join` field will synthesize [Result]s of the concurrent execution threads (spawned by `fork`), and incorporate them into the current execution thread.
+
+It is also possible for the thread to complete execution (omit `join` field) and leave concurrent threads behind. The `join` field, simply marks the execution thread and provides a way to follow it through trail of [Receipt]s.
+
+## 7.1 Schema
+
+```ipldsch
+# Represents a request to invoke enclosed set of tasks concurrently
+type Effect {
+  # Primary set of tasks to be invoked
+  fork      [&Invocation]
+  
+  # Additional task to be invoked with added semantics
+  # of representing a workflow execution continuation.
+  join       optional &Invocation
+}
+```
+
+## 7.2 Fields
+
+### 7.2.1 Forked Task Invocations
+
+The OPTIONAL `fork` field, if present MUST be a list of an alphabetically ordered [Invocation] links. List MUST NOT not contain duplicate entries. Every linked [Invocation] MUST have `cause` field set to the same [Invocation] (link) as the `run` field of the containing [Receipt].
+
+### 7.2.2 Joined Task Invocation
+
+The OPTIONAL `join` field, if present MUST be set to an [Invocation] link. Linked [Invocation] in the `join` field MUST have `cause` field set to the same [Invocation] (link) as the `run` field of the containing [Receipt].
+
+## 7.4 DAG-JSON Examples
+
+### 7.4.1 Effect spawning concurrent threads
+
+```json
+{
+  "fork": [
+    {
+      "/": "bafyreigtbbrfic7gduybn2lcncbbzycbef3kt3aqeguko5dmuhpmf73zmu"
+    },
+    {
+      "/": "bafyreidj2fkidh5g4tlldael7dfw27va6vbn6ah6pv7hirsandzwp25pj4"
+    }
+  ]
+}
+```
+
+### 7.4.2 Effect continuing thread execution
+
+```json
+{
+  "join": {
+    "/": "bafyreigqsmtpw6qliojfzupzbxsnkn4yave7vov2jvvrtye7w4nomvd7yq"
+  }
+}
+```
+
+### 7.4.1 Effect with fork & join
+
+```json
+{
+  "fork": [
+    {
+      "/": "bafyreieljernknkcrmefokxxperobictfndo4v4cazkjmqfwuog4puawxm"
+    },
+    {
+      "/": "bafyreiho3x2lbi6a5irw3orgrzrivkakr7iymalpukmxsvdsoqlxpcjsfq"
+    }
+  ],
+  "join": {
+    "/": "bafyreidtsxqd5fcuwdgverklwpcgbgzixsf6z6wgelybiakpwec3q75iwa"
+  }
+}
+```
+
+# 8 Receipt
+
+A `Receipt` is an attestation of the [Result] and a caused [Effect] by the [Task] [Invocation]. A Receipt MUST be signed by the [Executor] or it's delegate. If signed by the delegate, the proof of delegation from the [Executor] to the Issuer (the `iss` of the receipt) MUST be provided in `prf`.
```

Comment:
Minor wording change:

```suggestion
A `Receipt` is an attestation of the [Result] and requested [Effect]s by a [Task] or [Invocation]. A Receipt MUST be signed by the [Executor] or it's delegate. If signed by the delegate, the proof of delegation from the [Executor] to the Issuer (the `iss` of the receipt) MUST be provided in `prf`.
```

I still wonder if a recursive receipt is a cleaner solution here, but we can explore that in the next version